### DISCUSSION TITLE
Authenticating to grafana API with a managed identity

### DISCUSSION THREAD

**Discussion by Poltergeisen** [grafana/terraform-provider-grafana#1854] [Q&A] [answered]
We are using Azure Managed Grafana and I want to connect and manage dashboards with the API using a managed identity as a user. 

Is this possible?

**Comment by clementduveau** [comment, marked as answer]
That's a question for Azure's support.

**Reply by Poltergeisen** [reply]
Why is this a question for azure support? Wouldnt the Grafana provider just have to add a solution to get the tokens from their endpoint?

**Reply by clementduveau** [reply]
Azure implements Grafana in their own way to get their managed identity provider to work with Grafana, correct ?

**Reply by Poltergeisen** [reply]
Good question... I'm not really sure in this case. I made this question back when I was pretty new to cloud. I'm not sure how azure Grafana authentication would work and if they'd need to write a custom provider or create some way to authenticate that would be implemented here

**Reply by Duologic** [reply]
Grafana API [supports basic auth or tokens](https://grafana.com/docs/grafana/latest/developer-resources/api-reference/http-api/authentication/), an Azure specific authentication method should be asked through Azure support.

**Reply by Poltergeisen** [reply]
The more I learn about managed ID the more it makes sense at the application level. This isn't an azure concept, but a federated login concept. The api could support a msi token granted by azure, Google, k8s, wherever and as long as the oidc provider approves it'd work.

Is this still something MS would actually implement? Or wouldn't it be better implemented by the Grafana maintainers?

An example is fluxcd, a CNCF project, allowing managed ID login for passwordless work

https://fluxcd.io/flux/installation/configuration/workload-identity/
